### PR TITLE
Fix empty array when retrieving stats data

### PR DESCRIPTION
**Pull request by terryyylim**
**What this PR does / why we need it**:
These 2 methods, [this](https://github.com/feast-dev/feast/blob/master/core/src/main/java/feast/core/service/StatsService.java#L254) and [that](https://github.com/feast-dev/feast/blob/master/core/src/main/java/feast/core/service/StatsService.java#L185) will always be returning at least an empty arraylist. So essentially, when being called in [getFeatureStatistics](https://github.com/feast-dev/feast/blob/master/core/src/main/java/feast/core/service/StatsService.java#L92), these conditions, [this](https://github.com/feast-dev/feast/blob/master/core/src/main/java/feast/core/service/StatsService.java#L135) and [that](https://github.com/feast-dev/feast/blob/master/core/src/main/java/feast/core/service/StatsService.java#L151) will never be triggered, since length of arraylist will always at least be 1.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
N/A
```

### CODE REVIEW

File: core/src/main/java/feast/core/service/StatsService.java [hunk 1]
```diff
@@ -129,6 +129,9 @@ public GetFeatureStatisticsResponse getFeatureStatistics(GetFeatureStatisticsReq
         List<FeatureNameStatistics> featureNameStatistics =
             getFeatureNameStatisticsByDate(
                 statisticsRetriever, featureSet, features, timestamp, request.getForceRefresh());
+        if (featureNameStatistics.size() != 0) {
+          featureNameStatisticsList.add(featureNameStatistics);
```
Comment (woop) [inline core/src/main/java/feast/core/service/StatsService.java:133]: Why isnt this in `getFeatureNameStatisticsByDate`?

Comment (terryyylim) [reply core/src/main/java/feast/core/service/StatsService.java:133]: Would you mean returning null if the size is 0 in `getFeatureNameStatisticsByDate`?

**Pull request merged**